flask_static_digest/__init__.py

```diff
@@ -22,8 +23,23 @@ def init_app(self, app):
         :return: None
         """
         app.config.setdefault("FLASK_STATIC_DIGEST_BLACKLIST_FILTER", [])
-        app.config.setdefault("FLASK_STATIC_DIGEST_GZIP_FILES", True)
         app.config.setdefault("FLASK_STATIC_DIGEST_HOST_URL", None)
+        app.config.setdefault("FLASK_STATIC_DIGEST_GZIP_FILES", None)
+        app.config.setdefault("FLASK_STATIC_DIGEST_COMPRESSION", ["gzip"])
+
+        compression = app.config["FLASK_STATIC_DIGEST_COMPRESSION"] = set(
+            app.config["FLASK_STATIC_DIGEST_COMPRESSION"]
+        )
+
+        if app.config.get("FLASK_STATIC_DIGEST_GZIP_FILES") is not None:
+            warn(
+                "Configuration FLASK_STATIC_DIGEST_GZIP_FILES will be removed"
+                " in 0.5, please use FLASK_STATIC_COMPRESSION.",
+            )
+            if app.config.get("FLASK_STATIC_DIGEST_GZIP_FILES"):
+                compression.add("gzip")
```

Comment:
Can you change this and the discard item below this to:

```py
app.config["FLASK_STATIC_DIGEST_COMPRESSION"].add("gzip")
```

and:

```py
app.config["FLASK_STATIC_DIGEST_COMPRESSION"].remove("gzip")
```

Reply:
The logic will break if it isn't a set, we get two gzip's added into the list. Or we have to convert to and from a set a couple of times.

Reply:
Have a look at the latest one, I think it matches your style. I force pushed it to make it easier for you to see all the changes and to collapse all the tiny commits I did to fix `ci.yml` syntax and other errors.

I recommend you play with the option processing or do a quick gedankenexperiment in your head. I can't come up with a simple way to do it without keeping it a set for a while - if you think about it we have to juggle a default of `gzip` and two possible variables that add or remove `gzip`. You will always end up with `["gzip"]` or `[]` for valid configs when you want the opposite. To fix that 2+ `if ...` statements are then required, and existing statements need multiple conditions.

---
Architecture thoughts:

Personally, I believe a `set` is just better fundamental data-type for a set of options. You are not wrong that the later code would work with duplicates in a list, but that upsets my programming aesthetic and seems obviously worse that a set.

I am also struggling to see any benefits of converting it back to a list - with a config option with max. only two values performance and memory are not an issue. I can see a minor technical benefit to leave `app.config["FLASK_STATIC_DIGEST_COMPRESSION"]` as a sanitized list for other apps or the flask user, but we are still mutating its internal value from the users' config.

However, to maintain expectations that `app.config["FLASK_STATIC_DIGEST_COMPRESSION"]` is a list, I have changed the option processing to: `compression = set(...)`, processed the options leveraging set behavior for correctness, and then `app.config["FLASK_STATIC_DIGEST_COMPRESSION"] = list(compression)` at the end to stop the `set` leaking from inside `init_app(...)`.

Reply:
To be honest we could always drop the deprecation idea entirely and only support `FLASK_STATIC_DIGEST_COMPRESSION = ["gzip"]` as a list and ignore the whole set aspect.

I kind of feel like this idea of trying to support multiple defaults and list modifications based on an old setting is causing more pain than help. Realistically this project is pretty small and if anyone upgrades versions we can include a new ` - Remove FLASK_STATIC_DIGEST_GZIP_FILES config option` line item in the changelog.

Reply:
Since it is supported I propose leaving alone for now, and just delete it out of the next version as you proposed the first time. That would clean up that file easily.

Reply:
This would be changing this:

```py
        compression = set(app.config["FLASK_STATIC_DIGEST_COMPRESSION"])
        if app.config.get("FLASK_STATIC_DIGEST_GZIP_FILES") is not None:
            warn(
                "Configuration FLASK_STATIC_DIGEST_GZIP_FILES will be removed"
                " in 0.5, please use FLASK_STATIC_COMPRESSION.",
            )
            if app.config.get("FLASK_STATIC_DIGEST_GZIP_FILES"):
                compression.add("gzip")
            else:
                compression.discard("gzip")
        app.config["FLASK_STATIC_DIGEST_COMPRESSION"] = list(compression)
```

to:

```py
        compression = set(app.config["FLASK_STATIC_DIGEST_COMPRESSION"])
        app.config["FLASK_STATIC_DIGEST_COMPRESSION"] = list(compression)
```

Do you see any harm in doing that change as part of this PR? The alternative is we merge the PR and I make a separate commit a few minutes later after making a new release.